notifications/notifications.go

```diff
@@ -107,9 +100,10 @@ func (ps *impl) Subscribe(ctx context.Context, keys ...cid.Cid) <-chan blocks.Bl
 				if !ok {
 					return
 				}
+				// We could end up blocking here if the client
```

Comment:
I see that the waitgroup is the source of the race, and I get removing it, but I'm not 100% on why removing the cancel channel is needed, and it does seem like a nice way to insure these go routines get cleaned up, even if the subscribe client doesn't cancel context.

Reply:
I mostly just removed it because it wasn't absolutely necessary. We introduced it in the first place to avoid a lock, IIRC.

However, you're right. There's no reason _not_ to have it.